src/main/java/com/exacttarget/fuelsdk/ETClient.java

```diff
@@ -418,15 +495,13 @@ public synchronized String refreshToken()
             }
 
             logger.debug("refreshing access token...");
-
-            if (refreshToken == null) {
-                throw new ETSdkException("refreshToken == null");
-            }
         }
 
         requestToken(refreshToken);
 
-        soapConnection.setAccessToken(accessToken);
+        if(soapConnection != null) {
```

Comment:
why we are adding this null check?

Reply:
While doing tests involving accessToken expiration, I encountered a case where this code fails due to the soapConnection being null.